### PR TITLE
Merge release to main; Version bump to 1.0.43

### PR DESCRIPTION
**Pull request by WilliamXieMSFT**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,11 @@ Please format the changes as follows:
 + BugFixes:
 + Updates:
 
+## 1.0.42
+
 ## 1.0.41
++ Updates:
+  + Remove PathCch functions in order to support Win7
```
Comment (delmyers) [inline CHANGELOG.md:14]: I would wait until the other two PRs into release are finished.

**Review by delmyers — CHANGES_REQUESTED** [review]

**Review by delmyers — APPROVED** [review]

**Comment by WilliamXieMSFT** [comment]
Seems changes in main conflict with release:

../../InstrumentationEngine.Lib/libInstrumentationEngine.Lib.a(InstrumentationMethod.cpp.o): In function `MicrosoftInstrumentationEngine::CInstrumentationMethod::InitializeCore(bool)':
/root/ClrInstrumentationEngine/src/InstrumentationEngine.Lib/./../Common.Headers/StringUtils.h:74: undefined reference to `PathAppendW'
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Pull request merged**